frappe/model/naming.py

```diff
@@ -141,6 +140,9 @@ def parse_naming_series(parts, doctype='', doc=''):
 			part = today.strftime('%Y')
 		elif e == 'FY':
 			part = frappe.defaults.get_user_default("fiscal_year")
+		elif e.startswith('{') and doc:
+			e = e.replace('{', '').replace('}', '')
```

Comment:
This will also work

```
part = e.format(doc)
```

Reply:
This wouldn't work as the curly braces should be removed before replacing it with the field value